src/Orchard.Web/Modules/Orchard.MediaProcessing/Services/ImageProfileManager.cs

```diff
@@ -238,6 +256,30 @@ public class ImageProfileManager : IImageProfileManager {
             var filenameWithExtension = Path.GetFileName(path) ?? "";
             var fileLocation = path.Substring(0, path.Length - filenameWithExtension.Length);
 
+            // If absolute path is longer than the maximum path length (260 characters), file cannot be saved.
+            // File name is hashed to avoid homonyms when trimming similar file names.
+            var filenameWithoutExtension = Path.GetFileNameWithoutExtension(path).GetHashCode().ToString("x");
```

Comment:
`GetHashCode()` is not stable, it might return different values (even if it works locally most of the time).

You should use a stable hashing algorithm, like SHA256, MD5, ...

Reply:
I used GetHashCode() because it was already used to hash profile subfolders. The purpose is to avoid very large file and folder names; hashing with stable algorithms generates large strings (32 characters with MD5, 64 with SHA256). I may use them (and, for the same reason, I should use them to generate folder names too), but then I'd have to truncate those large strings to limit path length.